tests/ui/unicode-quote.stderr

```diff
@@ -0,0 +1,20 @@
+error: unknown character escape: `\u{201c}`
+  --> $DIR/unicode-quote.rs:2:26
+   |
+LL |     dbg!("since when is \“THIS\” not allowed in a string literal");
+   |                          ^ unknown character escape
+   |
+   = help: \u{201c} is not an ascii quote, but may look like one in some fonts.
+           consider writing it in its escaped form for clarity.
```

Comment:
Suggestion: the `help` message itself could look like

```
help: U+201C Left Double Quotation Mark (“) looks like U+0022 Quotation Mark (") (ASCII quotation mark) but are different characters
```

and a `MaybeIncorrect` suggestion as mentioned could look something like

```
help: consider writing the Unicode escape
  |
2 |     dbg!("since when is \u{201c}THIS\” not allowed in a string literal");
  |                         +++
```

(exact wording may vary)

Reply:
does rustc have a database of unicode character names?